### ISSUE TITLE
Disable click sound on navigating from a System.Windows.Controls.Frame

### ISSUE THREAD

**Issue by bbougot** [dotnet/wpf#2174]
I'm running a .NET Core 3.0 WPF in which a Frame is embedded into my main Window.
Every time I navigate in this frame, using its NavigationService, a disturbing click sound (the same one that is played while navigating in Internet Explorer or WebBrowser Control) occur and I considerate it as a *very bad* user experience.
As far as I know, there is no way to disable it, according to [this](https://stackoverflow.com/questions/4254026/turn-off-navigation-page-sound-in-wpf-frame-control) and [this](https://social.msdn.microsoft.com/Forums/vstudio/en-us/843677f4-8f0b-46cb-986c-92e8042d0707/stupid-problem-with-webbrowser-control?forum=vbgeneral).

I do not consider writing a registry value is a working solution here.

**Comment by miloush** [comment]
I have to agree with the links. If you consider it a _very bad_ user experience, turn the sound off in the settings. Other users might find it desirable, so user setting sounds a reasonable solution.

**Comment by bbougot** [comment]
I actually don't quite think it's a solution by any mean. Let's imagine you have another running application in the background (Netflix for instance), you'd be fine to make the decision turning off the sound only to avoid a design issue of WPF navigation?

**Comment by miloush** [comment]
Sorry that's not what I suggest, I suggest that if user does not want to hear navigation sounds, they need to go disable the "Start Navigation" sound in the sound settings. The point being, this is not a WPF specific experience - as you noted yourself it's generated by another system component.

**Comment by bbougot** [comment]
The sound has to be turned off in the OS settings which may not be very convenient. Anyway, it occurs only for Windows 7, and considering its usage is decreasing it won't be a problem in the upcoming years.

**Comment by miloush** [comment]
Since it's a user preference, it seems reasonable to me for the user to express their preference using an OS setting. I can even understand how an application trying to hide the fact that it's navigating could be seen as malicious. As you also noted, this sound has been removed completely in later OS versions.

I would consider this to be a WPF issue only if the NavigationService did not respect the existing setting.

**Comment by psmulovics** [comment]
Problem lies if you are trying to create an application that plays music / sound itself - eg you don't want Spotify to make funny noises while you navigate it

**Comment by miloush** [comment]
@psmulovics I think I don't see how that changes anything on the solution which is to disable the sound of the navigation event only. Your app can still produce any sounds it desires.

**Comment by bbougot** [comment]
From a user experience perspective, it still appears weird to me that the user has to change its OS settings only to *fix* an application behavior. But it's my sole point of view.

**Comment by weltkante** [comment]
I think the point is that not every instance of a `Frame` control represents a Browser-style experience. If you want an example where it is not appropriate: I would be quite annoyed if VS started to "honor" this OS setting and produce sounds on navigating source code. The App programmer should be able to opt-out of this behavior instead of having to rewrite the `Frame` control just for having the appropriate audio experience.

**Comment by miloush** [comment]
@bbougot my point was this is not an application-specific behaviour. You are setting it in OS for all frame navigation experience across the system. As I also said it might be a _fix_ for you but _fix_ for me might be to make sure there is a sound. If a developer could opt out, there is no way user could _fix_ it the other way.

I think I therefore disagree with @weltkante that an app programmer should be able to opt-out, but I don't have strong feelings either way. I was trying to say this is by design, there is a setting to control it by that design, and given that it has been the behaviour since beginning and nowadays only applicable in Windows 7 (resp. old IE), it's unlikely worth any investment, assuming it is actually within power of WPF.

That said, I wouldn't be arguing any PR related to this.
